src/components/panels/PanelsWrapper.vue

```diff
@@ -1,29 +1,71 @@
 <template>
-  <div class="panels-wrapper t-flex t-flex-col md:t-overflow-auto t-flex-1 md:t-flex-row md:t-space-x-4 t-space-y-4 md:t-space-y-0 t-px-4 lg:t-px-6 md:t-pb-4">
-    <Panel
-      v-for="(panel, i) in panels"
-      v-show="panel.show"
-      :key="`pc${i}`"
-      :panel="panel"
-      :active-view="getActiveView(i)"
-      class="t-flex-1 t-max-h-[500px] md:t-max-h-screen md:t-h-auto"
-      @active-view="onActiveViewChange($event, i)"
-    />
+  <div
+    class="panels-wrapper t-flex-1 t-space-y-4
+    md:t-space-y-0 t-px-4 lg:t-px-6 t-pb-2 lg:t-pb-4 t-overflow-y-hidden lg:t-overflow-hidden"
+  >
+    <div
+      ref="container"
+      class="t-flex t-h-full t-overflow-x-auto t-pb-2 lg:t-pb-0 -t-mr-4 lg:t-mr-0"
+    >
+      <Panel
+        v-for="(panel, i) in panels"
+        v-show="panel.show"
+        :key="`pc${i}`"
+        :panel="panel"
+        :active-view="getActiveView(i)"
+        :default-width="defaultWidth"
+        :class="{ 't-ml-4': i > 0 && !isMobile, 't-ml-2': i > 0 && isMobile, 't-mr-4': i === panels.length - 1 && isMobile }"
+        @active-view="onActiveViewChange($event, i)"
+      />
+    </div>
   </div>
 </template>
 
 <script setup>
 
-import { computed } from 'vue';
+import {computed, onMounted, onUnmounted, ref, useTemplateRef, watch} from 'vue';
 import { useConfigStore } from '@/stores/config';
 
 import Panel from '@/components/panels/Panel.vue';
+import { useResize } from "@/utils/resize.js";
+
+const { isMobile, onResize } = useResize();
 
 const configStore = useConfigStore();
 
+const panelWrapperPadding = 32;
+const panelMargin = 16;
+
 const panels = computed(() => config.value.panels);
 const config = computed(() => configStore.config);
 const activeViews = computed(() => configStore.activeViews);
+const container = useTemplateRef('container');
+const defaultWidth = ref(300)
+
+let unsubscribe;
+
+const updatePanelWidth = (isMobile) => {
+  if (container.value) {
+    defaultWidth.value = isMobile ? container.value.clientWidth - panelWrapperPadding : container.value.clientWidth / panels.value.filter(({ show }) => !!(show)).length - panelMargin;
+  }
+  return 300;
```

Comment:
Is this return statement needed ? I don't see whether it is assigned to some variable.

Reply:
If the container does not exist yet.